ddtrace/tracer/span.go

```diff
@@ -121,6 +121,16 @@ func (s *span) SetTag(key string, value interface{}) {
 		s.setMetric(key, v)
 		return
 	}
+	defer func() {
+		e := recover()
+		if e != nil {
```

Comment:
```suggestion
		if e := recover(); e != nil {
```
Maybe we can one-line this.

Also, for clarifications, two questions:
1. Can we thoroughly document why this recovery is here?
2. Should this `defer` not be added only inside the `if` condition below? Since it only treats a panic for the `fmt.Stringer` call. I don't think we need it in other cases.

Reply:
Good call on all of these.